samples/bot-feed-members/nodejs/bots/teamsConversationBot.js

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const {
+    ActionTypes,
+    CardFactory,
+    MessageFactory,
+    TeamsActivityHandler,
+    TeamsInfo,
+    TurnContext
+} = require('botbuilder');
+
+class TeamsConversationBot extends TeamsActivityHandler {
+    constructor() {
+        super();
+
+        this.onMessage(async (context, next) => {
+            // Code to get details of all members in group chat
+            const members = await TeamsInfo.getPagedMembers(context);
+            TurnContext.removeRecipientMention(context.activity);
+
+            const text = context.activity.text.trim().toLocaleLowerCase();
+            if (text.includes('list')) {
+                await this.listMembersAsync(context, members);
+            } else {
+                await this.cardActivityAsync(context, false);
+            }
+
+            await next();
+        });
+
+        this.onMembersAdded(async (context, next) => {
+                const members = await TeamsInfo.getPagedMembers(context);
+                var memberId = context.activity.membersAdded[0].aadObjectId;
+                var member = await TeamsInfo.getMember(context, memberId);
+                await context.sendActivity(`Welcome to the team :${member.name}`);
+                await this.listMembersAsync(context, members);
+                
+            await next();
+        });
+    }
+
+    async onInstallationUpdateActivity(context) {
+        if (context.activity.conversation.conversationType === 'channel') {
+            await context.sendActivity(MessageFactory.text(`Welcome to Microsoft Teams type list by @ mention the bot to get list of all members.`));
+        } else {
+            await context.sendActivity(MessageFactory.text('Welcome to Microsoft Teams type list by @ mention the bot to get list of all members.'));
```

Comment:
this IF statement has the same code on both conditions

Reply:
My understanding of requirements is: 
On member added -  display full info of the member added by calling get-member based on the id returned by the member added event. 
On installation update we should display the detailed  info of the user that installed the app (calling get member) and possibly also the info for all the other users   in the  chat by calling get-members-paged
Optionally, when user sends "list" command, bot should return the list of all users.

Reply:
Updated